sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -67,6 +68,46 @@ object HiveThriftServer2 extends Logging {
     }
   }
 
+  def runScript(sqlContext: HiveContext) {
+    val hiveConf = sqlContext.hiveconf
+    val script = hiveConf.get("hive.server2.init.script", null)
```

Comment:
I'm not super-familiar with this part of Hive, but a bit of sleuthing suggests that there's a `hive.server2.global.init.file.location` configuration for init scripts: https://cwiki.apache.org/confluence/display/Hive/Setting+Up+HiveServer2#SettingUpHiveServer2-OptionalGlobalInitFile

Why not use that?